File: src/main/java/com/seadowg/activity/TimerActivity.java
package com.seadowg.activity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.seadowg.R;

public class TimerActivity extends AppCompatActivity implements View.OnClickListener {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.timer);

    findViewById(R.id.start_button).setOnClickListener(this);
    findViewById(R.id.reset_button).setOnClickListener(this);
  }

  @Override
  public void onClick(View view) {
    switch (view.getId()) {
      case R.id.start_button:
        onSubmitWeight();
        break;
      case R.id.reset_button:
        onReset();
        break;
      default:
        break;
    }
  }

  private void onSubmitWeight() {
    try {
      Integer weight = Integer.parseInt(((TextView) findViewById(R.id.weight)).getText().toString());
      Integer totalMinsToCook = Math.round((weight / 500f) * 25f) + 25;
      Integer hoursToCook = totalMinsToCook / 60;
      Integer minsToCook = totalMinsToCook % 60;

      TextView remaining = (TextView) findViewById(R.id.remaining);
      remaining.setText(hoursToCook + ":" + minsToCook);

      findViewById(R.id.running).setVisibility(View.VISIBLE);
      findViewById(R.id.not_running).setVisibility(View.GONE);
    } catch (NumberFormatException e) {
      AlertDialog error = new AlertDialog.Builder(this)
              .setMessage(getString(R.string.no_weight_error))
              .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialogInterface, int i) {
                  dialogInterface.dismiss();
                }
              }).create();

      error.show();
    }
  }

  private void onReset() {
    findViewById(R.id.running).setVisibility(View.GONE);
    findViewById(R.id.not_running).setVisibility(View.VISIBLE);
  }
}

File: src/main/kotlin/com/seadowg/cluck/activity/TimerActivity.kt
package com.seadowg.cluck.activity

import android.os.Bundle
import android.view.View
import android.view.ViewGroup.MarginLayoutParams
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.updateLayoutParams
import com.seadowg.cluck.R
import com.seadowg.cluck.view.ErrorDialog
import com.seadowg.cluck.view.setUpEdgeToEdge
import kotlin.math.roundToInt

class TimerActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setUpEdgeToEdge(fitsSystemWindow = true)

        setContentView(R.layout.timer)

        val startButton = findViewById<Button>(R.id.start_button)
        startButton.setOnClickListener { onSubmitWeight() }
        val resetButton = findViewById<Button>(R.id.reset_button)

        resetButton.setOnClickListener { onReset() }
    }

    private fun onSubmitWeight() {
        try {
            val weight = Integer.parseInt((findViewById<TextView>(R.id.weight)).text.toString())
            val totalMinsToCook = ((weight / 450f) * 20f).roundToInt() + 10
            val hoursToCook = totalMinsToCook / 60
            val minsToCook = totalMinsToCook % 60

            val remaining = findViewById<TextView>(R.id.remaining)
            remaining.text = hoursToCook.toString() + ":" + "%02d".format(minsToCook)

            findViewById<View>(R.id.running).visibility = View.VISIBLE
            findViewById<View>(R.id.not_running).visibility = View.GONE
        } catch (e: NumberFormatException) {
            val error = ErrorDialog(this, getString(R.string.no_weight_error))
            error.show()
        }

    }

    private fun onReset() {
        findViewById<View>(R.id.running).visibility = View.GONE
        findViewById<View>(R.id.not_running).visibility = View.VISIBLE
    }
}
